3주차/[Lv.2]_숫자_변환하기/이현준.js

```diff
@@ -0,0 +1,22 @@
+function solution(x, y, n) {
+  const dp = new Array(y + 1).fill(Infinity);
+  dp[x] = 0;
+
+  for (let i = x; i <= y; i++) {
+    if (dp[i] === Infinity) continue;
+
+    if (i + n <= y) {
+      dp[i + n] = Math.min(dp[i + n], dp[i] + 1);
+    }
+
+    if (i * 2 <= y) {
+      dp[i * 2] = Math.min(dp[i * 2], dp[i] + 1);
+    }
+
+    if (i * 3 <= y) {
+      dp[i * 3] = Math.min(dp[i * 3], dp[i] + 1);
+    }
+  }
+
+  return dp[y] !== Infinity ? dp[y] : -1;
```

Comment:
```suggestion
  return isFinite(dp[y]) ? dp[y] : -1;
```
고차함수를 활용하면 코드가 훨씬 깔끔할 것 같아서 적어보았습니다! 😆